ddl/reorg.go

```diff
@@ -168,13 +204,23 @@ func (w *worker) runReorgJob(t *meta.Meta, reorgInfo *reorgInfo, tblInfo *model.
 		logutil.BgLogger().Info("[ddl] run reorg job quit")
 		w.reorgCtx.setNextHandle(nil)
 		w.reorgCtx.setRowCount(0)
+		w.reorgCtx.resetWarnings()
 		// We return errWaitReorgTimeout here too, so that outer loop will break.
 		return errWaitReorgTimeout
 	case <-time.After(waitTimeout):
 		rowCount, doneHandle := w.reorgCtx.getRowCountAndHandle()
 		// Update a job's RowCount.
 		job.SetRowCount(rowCount)
 		updateAddIndexProgress(w, tblInfo, rowCount)
+
+		// Update a job's warnings.
+		w.reorgCtx.mu.Lock()
+		partWarnings := w.reorgCtx.mu.warnings
+		partWarningsCount := w.reorgCtx.mu.warningsCount
+		job.SetWarnings(mergeWarningsAndWarningsCount(partWarnings, job.ReorgMeta.Warnings, partWarningsCount, job.ReorgMeta.WarningsCount))
+		w.reorgCtx.mu.Unlock()
+
+		w.reorgCtx.resetWarnings()
```

Comment:
Why we need to reset warnings here？

Reply:
It's a partial result. After it was merged into the job, the next round DDL will collect new ones.